### ISSUE TITLE
Console looks is just looping continuously because "broadcastMessage is not callableEmbedded SWF (flash) into html doesn't run in Ruffle.

### ISSUE THREAD

**Issue by winston895** [ruffle-rs/ruffle#2301]
Embedded SWF (flash) into html doesn't run in Ruffle, and it was suggested that I post the file(s) here for a developer to look at. (may be the missing puzzle to get Ruffle being more compatible. )

**Comment by winston895** [comment]
[FluidsandElectrolytes.zip](https://github.com/ruffle-rs/ruffle/files/5774343/FluidsandElectrolytes.zip)

**Comment by winston895** [comment]
The FluidsandElectrolytes.zip file contains 1 html file + 1 folder (which contains 6 files including the SWF file).

**Comment by Poopooracoocoo** [comment]
@winston895 did you mean to close your issue btw?

**Comment by winston895** [comment]
Sorry, I did not mean to close the issue. Any help would be appreciated.

**Comment by EmperorBale** [comment]
are there any updates on this issue?

**Comment by Herschel** [comment]
Haven't looked at this deeply yet, but the message itself is not necessarily a problem -- in fact, we've disabled warning logs by default in recent web builds. Calling a non-existent method happens quite often in AVM1 content. Specifically the Flash built-in BroadcasterMX component lazily initializes the broadcastMessage method, so broadcastMessage doesn't exist, which is totally fine -- the call is just a no-op. So the warning message is red herring.

I need to investigate this more, but I suspect this particular instance is another case of #1535. The AS execution stops because of a goto, when it shouldn't because the clip should be kept alive by an unload handler.

In DefineSprite 10009:
```
      if(percent > 99.5)
      {
          // This gotoAndPlay causes execution to stop in Ruffle, but it doesn't in FlashP layer
         _parent._parent.gotoAndPlay(2);
         // Execution does not continue here
         gotoAndStop("off");
         doneLoading = 1;
      }
```

**Comment by CUB3D** [comment]
TLDR: This isn't due to #1535, the header in this swf tells lies.

According to the header in this SWF `frameCount = 2`, in reality there are 77 frames.
When the enterFrame event is dispatched to sprite 10009, it invokes `preload()`, this function contains the code from @Herschel comment  above, which moves the stage to frame 2. After executing frame 2, in Flash Player it then moves to frame 3, however in Ruffle it assumes that there are no more frames because it trusts the header and it loops back to frame 1.

Here's a basic example, runs 3 frames in Flash but only 2 in Ruffle: 
[trust_nothing.zip](https://github.com/ruffle-rs/ruffle/files/10283645/trust_nothing.zip)

And here is a *heavily* cut down version of the linked swf with debug traces, loops in Ruffle, doesn't in Flash: 
[fluids2.zip](https://github.com/ruffle-rs/ruffle/files/10283652/fluids2.zip)

Temporarily patching Ruffle to ignore the header (annoyingly frame count is one of the few header fields that JPEXS can't edit), makes it sort-of work: 
![image](https://user-images.githubusercontent.com/3901587/209051576-e9be02cd-969e-4d94-b342-46be5e5bc3da.png)

I'm guessing we will have to ignore the frame count in the header and check ahead for any extra frames before looping back

**Comment by Lord-McSweeney** [comment]
Fixed by #21526
